src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+	m_current_iteration = 0;
+	m_complete = false;
+
+	SG_ADD(
+	    &m_current_iteration, "current_iteration",
+	    "Current Iteration of training", MS_NOT_AVAILABLE);
+	SG_ADD(
+	    &m_max_iterations, "max_iterations", "Maximum number of Iterations",
+	    MS_AVAILABLE);
+	SG_ADD(&m_complete, "complete", "Convergence status", MS_NOT_AVAILABLE);
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+}
+
+bool CIterativeLinearMachine::train_machine(CFeatures* data)
+{
+	m_current_iteration = 0;
+	m_complete = false;
+	init_model(data);
+	return continue_train();
+}
+
+bool CIterativeLinearMachine::continue_train()
+{
+	reset_computation_variables();
```

Comment:
``reset_computation_variables`` reset the flags used to prematurely stop the computations (see the `StoppableSGObject` class). If you do not call this and you have prematurely stopped the execution, the next time you will call `train`/`continue_train` those methods won't work because of COMPUTATION_CONTROLLERS (since the flag is still set to false).

Reply:
I see. Maybe rename and document

Reply:
or maybe I should rather read :)